### PR TITLE
Minor refactoring

### PR DESCRIPTION
**Pull request by gausnes**
* Updated to use environment variable for InstalledSmartAppId
* Including smartthings_rsa on slash command handling
* smartapp to smartApp
* [(await).catch to try/catch](https://javascript.info/async-await#error-handling)
* Fixed missing and extra `;`s

### CODE REVIEW

File: src/README.md [hunk 1]
```diff
@@ -26,7 +26,7 @@ server.post('/', (req, res) => {
   smartApp.handleHttpCallback(req, res);
 });
 
-server.post('/slashCommand', async (req, res) => {
+server.post('/slack', async (req, res) => {
```
Comment (gausnes) [inline src/README.md]: This example code doesn't match up with the exposed endpoints of the included code

File: src/lib/slack/commands/SlashCommand.js [hunk 2]
```diff
@@ -26,26 +26,27 @@ const SlashCommand = {
...
+};
 
 function requireParams(body, params) {
-    var validated = true;
-    for (var param of params) {
+    for (const param of params) {
         if (!body[param]) {
             console.error(`ERROR ** Unable to init SlashCommand, no ${param} param found in body:`, body);
-            validated = false;
+            return false;
```
Comment (ajguyot) [inline src/lib/slack/commands/SlashCommand.js]: Nice, for some reason I thought `return` did not break `for` loops.

File: src/lib/slack/commands/ThingsBotCommand.js [hunk 3]
```diff
@@ -19,18 +19,18 @@ const ThingsBotCommand = {
...
                     "text": "Uh oh, there was a problem pulling your device list. Please try again."
                 };
-            } else if (deviceList.length == 0) {
+            } else if (deviceList.length === 0) {
                 return {
                     "response_type": "ephemeral",
                     "text": "Oops, looks like you don't have any devices on your SmartThings account! Please add a device and then try again."
                 };
             }
-
```
Comment (ajguyot) [inline src/lib/slack/commands/ThingsBotCommand.js]: Thanks. I think Sublime Text is more on top of stripping unnecessary whitespace. I'm tentatively using VS Code now and it doesn't seem to be as good about that. Also Sublime shows whitespace when you select lines which is really helpful. Code only has a global option to always show it or never show it so I have that disabled because I don't like the clutter always there

File: src/lib/slack/commands/ThingsBotCommand.js [hunk 4]
```diff
@@ -19,18 +19,18 @@ const ThingsBotCommand = {
...
 
             if (!deviceList) {
-                console.error('ERROR ** Failed to pull valid device list in deviceCommandsBlock getter')
+                console.error('ERROR ** Failed to pull valid device list in deviceCommandsBlock getter');
                 return {
                     "response_type": "ephemeral",
                     "text": "Uh oh, there was a problem pulling your device list. Please try again."
                 };
-            } else if (deviceList.length == 0) {
+            } else if (deviceList.length === 0) {
```
Comment (ajguyot) [inline src/lib/slack/commands/ThingsBotCommand.js]: Any particular reason for switching to triple equals here? `.length` is always going to return a `Number`, so allowing type conversion isn't an issue. Also if for some reason it returned something that converted to `0` I think we'd still want to trip the no devices response rather than sending a broken normal response.

File: src/lib/slack/index.js [hunk 5]
```diff
@@ -6,28 +6,29 @@ const router = require('express').Router(),
...
-    .catch((err) => {
+    try {
+        await promise
+    } catch (err) {
         console.error('ERROR ** Caught unhandled error in API execution:', err);
-    });
+    }
 
-    if (process.env.EXECUTION_CONTEXT == 'LAMBDA') {
+    if (process.env.EXECUTION_CONTEXT === 'LAMBDA') {
```
Comment (ajguyot) [inline src/lib/slack/index.js]: Same question on triple equals. Nothing type-converts to a particular string.

File: src/lib/slack/commands/ThingsBotCommand.js [hunk 6]
```diff
@@ -71,15 +71,12 @@ const ThingsBotCommand = {
...
         })();
     }
-}
+};
 
 async function getDeviceList(installedSmartAppId) {
-    const [ contextError, smartappContext ] = await smartappService.getSmartAppContext(installedSmartAppId);
-    if (contextError) { return; }
+    const smartAppContext = await smartAppService.getSmartAppContext(installedSmartAppId);
+    const deviceList = await smartAppService.getDeviceList(smartAppContext);
```
Comment (ajguyot) [inline src/lib/slack/commands/ThingsBotCommand.js]: By stripping the error handling here and throwing an error for problems getting context in `smartAppService`, aren't we just going to leave the user hanging with no error response? Or is that error going to get caught somewhere so the user gets updated that there was a problem?

File: src/lib/slack/commands/SlashCommand.js (hunk 2 above)
Comment (gausnes) [reply src/lib/slack/commands/SlashCommand.js]: That could be true, I can do some validation there. I can't say the my JS is that great

File: src/lib/slack/commands/ThingsBotCommand.js (hunk 3 above)
Comment (gausnes) [reply src/lib/slack/commands/ThingsBotCommand.js]: Just WebStorm things :)

File: src/lib/slack/commands/ThingsBotCommand.js (hunk 4 above)
Comment (gausnes) [reply src/lib/slack/commands/ThingsBotCommand.js]: WebStorm was throwing some yellow my way so I just made the recommended edit, I can remove tho. I was just listening to the tooling here

File: src/lib/slack/index.js (hunk 5 above)
Comment (gausnes) [reply src/lib/slack/index.js]: Same as above as well

File: src/lib/slack/commands/ThingsBotCommand.js (hunk 6 above)
Comment (gausnes) [reply src/lib/slack/commands/ThingsBotCommand.js]: Ya I think I have to visit this a bit, I was thinking that if we wanted an upstream call to handle errors they should try and catch these method invocations explicitly.

Again just a stylistic change here that felt more familiar to me
